src/main/java/net/neoforged/neoforge/attachment/IAttachmentSyncHandler.java

```diff
@@ -0,0 +1,17 @@
+package net.neoforged.neoforge.attachment;
+
+import net.minecraft.network.RegistryFriendlyByteBuf;
+import net.minecraft.server.level.ServerPlayer;
+import org.jetbrains.annotations.Nullable;
+
+/**
+ * Manages how data attachments are written (on the server) and read (on the client) from packets.
+ */
+public interface IAttachmentSyncHandler<T> {
+    // TODO: pass target player
+    void write(RegistryFriendlyByteBuf buf, T attachment, ServerPlayer to, AttachmentSyncReason reason);
```

Comment:
The player should not be context to the `write` method. Network synchronized data does not transform itself based on the target player, and we should not carve an exception to that rule here.